app/services/content_data_url.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+class ContentDataUrl
+  # department-for-work-pensions, department-of-health-and-social-care, government-digital-service
+  CONTENT_DATA_BETA_PARTNERS = %w(b548a09f-8b35-4104-89f4-f1a40bf3136d
+                                  7cd6bf12-bbe9-4118-8523-f927b0442156
+                                  af07d5a5-df63-4ddc-9383-6a666845ebe9).freeze
+
+  attr_reader :document
+
+  def initialize(document)
+    @document = document
+  end
+
+  def url
+    content_data_root = Plek.new.external_url_for("content-data-admin")
+    content_data_root + "/metrics" + document.current_edition.base_path
+  end
+
+  def displayable?(user)
+    has_content_data_access?(user) && expected_in_content_data?
+  end
+
+private
+
+  def has_content_data_access?(user)
+    CONTENT_DATA_BETA_PARTNERS.include?(user.organisation_content_id)
+  end
+
+  def expected_in_content_data?
+    published_before_yesterday? ||
+      (published_yesterday? && content_data_etl_end_time_elapsed?)
+  end
+
+  def published_yesterday?
+    document.first_published_at.to_date == Date.yesterday
+  end
+
+  def published_before_yesterday?
+    document.first_published_at.to_date < Date.yesterday
+  end
+
+  def content_data_etl_end_time_elapsed?
+    Time.current >= Time.current.change(hour: 9)
```

Comment:
Minor, but this could be `Time.current.hour >= 9`

I think it'd be useful to have a small comment that explains the oddness of this 9am and the 404 as I think it's a high enough WTF that you'd want it documenting beyond a commit message. Also worth stating we expect them to fix this so it can be removed.

Reply:
Added a comment 👍